core/src/kura.rs

```diff
@@ -404,7 +404,13 @@ impl<IO: DiskIO> BlockStore<IO> {
         #[allow(clippy::cast_possible_truncation)]
         buffer.resize(len as usize, 0);
         let _len = file_stream.read_exact(&mut buffer).await?;
-        Ok(Some(VersionedCommittedBlock::decode_versioned(&buffer)?))
+
+        let mut res = VersionedCommittedBlock::decode_all_versioned(&buffer);
+        if let Err(iroha_version::error::Error::ExtraBytesLeft(left)) = res {
```

Comment:
This is a frequently recurring pattern. I'd consider extracting it into a function.

Reply:
I'll use a macro. It will be more compact